src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,92 @@
-import { useState } from 'react';
+import React, { FormEvent, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd:(movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [formError, setFormError] = useState(true);
+
+  const handleInput = (value: string, setValue: (value: string) => void) => {
+    setValue(value);
+
+    if (title.trim() && imgUrl.trim() && imdbUrl.trim() && imdbId.trim()) {
+      setFormError(false);
+    }
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    setCount(count + 1);
```

Comment:
here should use callback form too )
https://mate.academy/learn/react/updating-state#/video/318
```suggestion
    setCount(count => count + 1);
```